Add upgrade note for routing rest migration (#8011)
sulu/sulu: UPGRADE-2.x.md

# Upgrade

## 2.6.10

### Deprecate usage of fos rest routing

> We are not longer considering the [fos rest routing](https://github.com/handcraftedinthealps/RestRoutingBundle) as a best practice.

Reviewer: ```suggestion
We are no longer considering the [fos rest routing](https://github.com/handcraftedinthealps/RestRoutingBundle) as a best practice.
```

All bundles should use the Symfony routing system instead.

> Inside your `config/routes/sulu_admin.yaml` and `config/routes/sulu_website.yaml` you can remove the fos rest routing.

Reviewer: ```suggestion
Inside your `config/routes/sulu_admin.yaml` and `config/routes/sulu_website.yaml`, you can remove the fos rest routing configuration.
```

> By first remove all `type: rest` appeareance and also replace `.yml` with `.yaml`:

Reviewer: ```suggestion
First, remove all appearances of `type: rest` and also replace `.yml` with `.yaml`:
```

Reviewer 2: appearances -> occurrences / instances

> First, remove all appearances of `type: rest` and also replace `.yml` with `.yaml`:

Reviewer: ```suggestion
First, remove all instances of `type: rest` and also replace `.yml` with `.yaml`:
```

```diff
sulu_tag_api:
-    resource: "@SuluTagBundle/Resources/config/routing_api.yml"
+    resource: "@SuluTagBundle/Resources/config/routing_api.yaml"
-    type: rest
     prefix: /admin/api
```

> This is an example and all routes should be checked for the `type: rest` and `.yml`.

Reviewer: ```suggestion
This is an example, all routes should be checked for the use of `type: rest` and `.yml`.
```

Reviewer 2: Maybe we could explicitly say to check that those routes should **not** use `type:rest`

> All `.yml` routing files of the core bundles will in future be removed.

Reviewer: ```suggestion
All `.yml` routing files of the core bundles will be removed in the future.
```

> If you want upgrade your own custom routes have a look at the command opt out command of [RestRoutingBundle](https://github.com/handcraftedinthealps/RestRoutingBundle).

Reviewer: ```suggestion
If you want to upgrade your own custom routes, have a look at the opt out command of [RestRoutingBundle](https://github.com/handcraftedinthealps/RestRoutingBundle).
```
